### PR TITLE
[server] report internal errors to check permissions

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

Right now we confuse users that they don't have permissions when we fail to check permissions.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 63ecd98</samp>

Improved error handling for SpiceDB authorization in `server` component. Added `ApplicationError` class to `spicedb-authorizer.ts` file.

</details>

## Related Issue(s)

Fixes #

## How to test

- Log in to perform FGA migration.
- Down scale spicedb pods.
- Try to perform some actions.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ak-fga-rep4b752e8dfa</li>
	<li><b>🔗 URL</b> - <a href="https://ak-fga-rep4b752e8dfa.preview.gitpod-dev.com/workspaces" target="_blank">ak-fga-rep4b752e8dfa.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ak-fga_report_internal_failed_checks-gha.18853</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ak-fga-rep4b752e8dfa%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/authorization/spicedb-authorizer.ts [hunk 1]
```diff
@@ -110,7 +111,7 @@ export class SpiceDBAuthorizer {
                 log.error("[spicedb] Failed to perform authorization check.", err, {
                     request: new TrustedValue(req),
                 });
-                return { permitted: !featureEnabled };
+                throw new ApplicationError(ErrorCodes.INTERNAL_SERVER_ERROR, "Failed to perform authorization check.");
```
Comment (svenefftinge) [inline components/server/src/authorization/spicedb-authorizer.ts:114]: If `!featureEnabled` we should return `permitted = true` and only throw if the feature is enabled

Comment (akosyakov) [reply components/server/src/authorization/spicedb-authorizer.ts:114]: makes sense, but from code it looks like already a case, since we don't await?

**Review by svenefftinge — APPROVED** [review]

**Comment by akosyakov** [comment]
that how it looks like now
Screenshot 2023-10-20 at 14 45 50

**Comment by akosyakov** [comment]
I guess we don't have request id in preview envs because tracing is off somehow?

**Comment by svenefftinge** [comment]
It could also be that we we didn't create a context before we looked up the user.

**Pull request merged**